### ISSUE TITLE
Make this editor full screen. 

### ISSUE THREAD

**Issue by ats1999** [nhn/tui.editor#1284]
## Describe the bug
While using the method `toolbar.insertItem` i am getting this error. 
```js
TypeError
this.el.appendChild is not a function
```

## To Reproduce
1. Go to https://codesandbox.io/s/fragrant-hooks-gr651?file=/src/App.js:393-411 and https://github.com/nhn/tui.editor/issues/1245
2. See an error

## Expected behavior
If I am returning the wrong value then, It should not rise `this.el.appendChild is not a function`. Because `this. el` is a class member.
 see this https://github.com/nhn/tui.editor/blob/3a0e0a93ae1970c2232060bf3ff747dd22413563/apps/editor/src/js/ui/uicontroller.js#L61. 

If i am doing anything wrong then please suggest me.

**Comment by seonim-ryu** [comment]
@ats1999 Not just assigning a function, but passing it by executing it like this..

```js
function createButton() {
  const el = document.createElement("button");

  el.innerHTML = "F";

  return el;
}

// ...
useEffect(() => {
  if (editorRef && editorRef.current) {
    const toolbar = editorRef.current.getInstance().getUI().getToolbar();
    toolbar.insertItem(0, {
      type: "button",
      options: {
        className: "full-screen",
        event: "test",
        tooltip: "Full screen",
        text: "",
        el: createButton(), // this
        style: "",
      },
    });
  }
}, []);
```

And you can apply the style directly to the button text.

![wrong](https://user-images.githubusercontent.com/18183560/101585415-0f4e7b00-3a23-11eb-8063-0d7dcb7469d3.png)

**Comment by ats1999** [comment]
It was there, but it was not visible until I passed `color: black`. Now, it's visible.

**Comment by ats1999** [comment]
I did this to make the editor full screen. 
```js
 function createButton() {
        const el = document.createElement("button");

        el.innerHTML = "F";
        el.onclick = (e)=>{
            toggleEditorFullScreen(editorRef.current.getRootElement());
        }
        return el;
    }
```

```js
export const toggleEditorFullScreen = (element ) =>{
        if (element.requestFullscreen) {
          element.requestFullscreen();
        } else if (element.mozRequestFullScreen) {
          element.mozRequestFullScreen();
        } else if (element.webkitRequestFullscreen) {
          element.webkitRequestFullscreen(Element.ALLOW_KEYBOARD_INPUT);
        } else if (element.msRequestFullscreen) {
          element.msRequestFullscreen();
        }
}
```

## See what i got
![Screenshot (495)](https://user-images.githubusercontent.com/54087826/101591031-a1fd1300-3a3f-11eb-9ccd-7ed5e5afe240.png)


I tried to make a parent container for editor like
```js
<div id="editor">
<Editor ..props/>
</div>
```
It also didn't worked.

**Comment by yar2001** [comment]
Here is what I had coded and it can work.
(Typescript is used, but that's the same)
```
const editorHeight = '600px';

function toggleFullScreen(element: HTMLElement) {
    const style = element.style;
    if (style.position === 'fixed') {
        style.position = '';
        style.top = '';
        style.left = '';
        style.zIndex = '';
        style.width = '';
        style.height = editorHeight;
    } else {
        console.log(style);
        style.position = 'fixed';
        style.top = '0px';
        style.left = '0px';
        style.zIndex = '10050';
        style.width = '100% !important';
        style.height = '100% !important';
    }
}

function Editor() {
    const editorRef: React.RefObject<ToastEditor> = useRef(null);
    useEffect(() => {
        const editor = editorRef.current?.getInstance();
        const eventManager = (editor as any).eventManager;
        eventManager.addEventType('toggleFullScreen');
        eventManager.listen('toggleFullScreen', function () {
            toggleFullScreen(
                editorRef.current?.getRootElement() as HTMLElement
            );
        });
    });
    return (
        <>
            <Card banner={{ id: 7, title: '枝条' }}>
                <ToastEditor
                    ref={editorRef}
                    initialValue="hello react editor world!"
                    initialEditType="wysiwyg"
                    language="zh-CN"
                    height={editorHeight}
                    usageStatistics={false}
                    onChange={() => {}}
                    toolbarItems={[
                        'heading',
                        'bold',
                        'italic',
                        'strike',
                        'divider',
                        'hr',
                        'quote',
                        'divider',
                        'ul',
                        'ol',
                        'task',
                        'divider',
                        'table',
                        'image',
                        'link',
                        'divider',
                        'code',
                        'codeblock',
                        {
                            type: 'button',
                            options: {
                                event: 'toggleFullScreen',
                                tooltip: '全屏',
                                text: 'F'
                            }
                        }
                    ]}
                />
            </Card>
        </>
    );
}
```

**Comment by ats1999** [comment]
But, this will make the editor full screen in only browser window. I want to make full screen in pc screen.

It is working fine,but css style is breaking.

**Comment by yar2001** [comment]
> But, this will make the editor full screen in only browser window. I want to make full screen in pc screen.
> 
> It is working fine,but css style is breaking.

Oh sure, you can customize it to your liking

**Comment by ats1999** [comment]
I did, what i want. 

This is what didn't worked.

**Comment by yar2001** [comment]
> I did, what i want.
> 
> This is what didn't worked.

What? I changed the code to the following, and it worked. Maybe I don't get what you mean.
```
function toggleFullScreen(element: HTMLElement) {
    element.requestFullscreen();
}
```

**Comment by ats1999** [comment]
I have added screen shot of editor when it becomes full screen.

You can see when the editor becomes full screen then preview part is becoming dark.

**Comment by yar2001** [comment]
> I have added screen shot of editor when it becomes full screen.
> 
> You can see when the editor becomes full screen then preview part is becoming dark.

I saw it! The background color becomes the background color of my website. Maybe when the editor is in preview mode, the div will be transparent . My solution is to add CSS(or simply use WYSIWYG mode):
```
.tui-editor .te-preview-style-tab > .te-tab-active {
    background: white;
}
```

**Comment by ats1999** [comment]
>My solution is to add CSS(or simply use WYSIWYG mode

WYSIWYG mode doesn't support code highlighting. So, i need to use preview mode.

But, I'll try adding css.

**Comment by ats1999** [comment]
> > I have added screen shot of editor when it becomes full screen.
> > You can see when the editor becomes full screen then preview part is becoming dark.
> 
> I saw it! The background color becomes the background color of my website. Maybe when the editor is in preview mode, the div will be transparent . My solution is to add CSS(or simply use WYSIWYG mode):
> 
> ```
> .tui-editor .te-preview-style-tab > .te-tab-active {
>     background: white;
> }
> ```

this also didn't worked.
```
.tui-editor .te-preview-style-tab > .te-tab-active {
    background: white;
}

```

**Comment by ats1999** [comment]
Now, i tried to give this styling to the root element. 
![Screenshot (497)](https://user-images.githubusercontent.com/54087826/101675747-4e75de00-3aae-11eb-8856-f7c33a88d0b1.png)
## Output

![Screenshot (496)](https://user-images.githubusercontent.com/54087826/101675768-56ce1900-3aae-11eb-937e-455c01a700f4.png)

### Problem
background contents are also visible in markdown mode(in preview part only). There is no problem in WYSIWYG mode.

**Comment by ats1999** [comment]
this worked
```
 const toggleEditorFullScreen = (el) =>{
    if(el.style.height !== "100vh")
      el.style = "height:100vh; width:100vw; position:fixed;z-index:10000000000;top:0px;left:0px;background-color:white;";
    else el.style = "height:400px;"
  }
```
